Google.Cloud.Spanner.NHibernate.Tests/NHibernateMockServerTests.cs

```diff
@@ -0,0 +1,139 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Google.Cloud.Spanner.Connection.MockServer;
+using Google.Cloud.Spanner.NHibernate.Tests.Entities;
+using Google.Cloud.Spanner.V1;
+using Google.Protobuf.WellKnownTypes;
+using NHibernate;
+using Xunit;
+
+namespace Google.Cloud.Spanner.NHibernate.Tests
+{
+    /// <summary>
+    /// Tests CRUD operations using an in-mem Spanner mock server.
+    /// </summary>
+    public class NHibernateMockServerTests : IClassFixture<NHibernateMockServerFixture>
```

Comment:
nit: I think it is good that you chose to create a separate test class for this. But would you mind changing the name of the test class to something like NHibernateVersioningMockServerTests? And also update the comment above the test class.